### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Graycemuthui**
### Description
In this project I achieved the following:
Remove current_user method in ApplicationController, devise will provide us with one.
[Install and setup devise](https://github.com/heartcombo/devise#getting-started).
Can register a new user.
User logs in with a combination of email and password.
Hashed password should be stored in the database.
Ask for confirmation of email.
Can reset password.

### Changes made
Add email confirmation

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
 class ApplicationController < ActionController::Base
-  def current_user
```
Comment (mateo951) [inline app/controllers/application_controller.rb:2]: current_user method in ApplicationController has been removed 😎 Well done!

File: app/models/user.rb [hunk 2]
```diff
@@ -1,10 +1,15 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (mateo951) [inline app/models/user.rb:4]: The application allows for registering a new user ✔

Comment (mateo951) [reply app/models/user.rb:4]: Please be sure to add email confirmation to your app as it is part of the requirements 📝

Comment (mateo951) [reply app/models/user.rb:4]: A user can reset the password in the application 🙌 Great!

**Review by mateo951 — APPROVED** [review]
### Hi @Graycemuthui, 

**🔥Highlights🔥**
- User can register, reset password, login ✔

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Comment by mateo951** [comment]
Sorry 😥 I unintentionally marked it as approved. Please make the mentioned changes so that you can ask for another review 🐐

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Graycemuthui 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**